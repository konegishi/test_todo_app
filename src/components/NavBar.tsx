import React from 'react';
import { supabase } from '../lib/supabase';

/**
 * NavBarのProps
 */
interface NavBarProps {
  isTransparent?: boolean;
}

/**
 * 指標ごとのカードコンポーネント
 * @param props NavBarのProps
 * @returns NavBarコンポーネント
 */
const NavBar: React.FC<NavBarProps> = () => {
  return (
    <React.Fragment>
      <nav className='bg-white shadow top-0 left-0 right-0'>
        <div className='container px-4 py-2 mx-auto'>
          <div className='flex items-center justify-between'>
            <span>ToDo Apps</span>
            <div className='flex'>
              <button
                className='fas fa-sign-out-alt h-5 w-5 ml-auto mr-2 text-gray-400'
                onClick={async () => {
                  const { error } = await supabase.auth.signOut();
                  // eslint-disable-next-line no-console
                  if (error) console.log('Error logging out:', error.message);
                }}
              />
            </div>
          </div>
        </div>
      </nav>
    </React.Fragment>
  );
};

export default NavBar;
