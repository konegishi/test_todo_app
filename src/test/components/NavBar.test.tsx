import { render } from '@testing-library/react';
import React from 'react';
import NavBar from '../../components/NavBar';

describe('NavBar component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<NavBar />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
