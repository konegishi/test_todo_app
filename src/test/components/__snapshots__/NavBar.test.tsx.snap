// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-white shadow top-0 left-0 right-0"
  >
    <div
      class="container px-4 py-2 mx-auto"
    >
      <div
        class="flex items-center justify-between"
      >
        <span>
          ToDo Apps
        </span>
        <div
          class="flex"
        >
          <button
            class="fas fa-sign-out-alt h-5 w-5 ml-auto mr-2 text-gray-400"
          />
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
